src/main/java/io/supertokens/httpRequest/HttpRequest.java

```diff
@@ -152,12 +152,12 @@ public static <T> T sendGETRequestWithResponseHeaders(Main main, String requestI
             if (version != null) {
                 con.setRequestProperty("api-version", version + "");
             }
-
+            con.setInstanceFollowRedirects(false);
             int responseCode = con.getResponseCode();
 
             con.getHeaderFields().forEach((key, value) -> {
                 if (key != null) {
-                    responseHeaders.put(key, value.get(0));
+                    responseHeaders.put(key, value.get(0)); // TODO why the first element only? What happens with Set-Cookie headers? (Those are repeated if there are multiple cookies)
```

Comment:
Answering the TODO: This is a bug in the existing code. Fixing this would also mean that we can't use `key` in the map, cause value.get(1) would override the previous entry in the key. So it would perhaps need to change to map(string, string[])